app/src/main/java/org/mozilla/fenix/home/pocket/PocketStoriesComposables.kt

```diff
@@ -150,6 +182,52 @@ fun PocketStories(
     }
 }
 
+private fun Modifier.onShown(
+    @FloatRange(from = 0.0, to = 1.0) threshold: Float,
+    onVisible: () -> Unit,
+): Modifier {
+    return composed {
+        val context = LocalContext.current
+        var wasEventReported by remember { mutableStateOf(false) }
+
+        onGloballyPositioned { coordinates ->
+            @Suppress("CollapsibleIfStatements")
+            if (!wasEventReported) {
+                if (coordinates.isVisible(context, threshold)) {
```

Comment:
how come the `@Suppress` here?

Reply:
When trying different approaches separate `if`s were easier to reason about. Will join them together in the final version.